### ISSUE TITLE
Your_Algorithm

### ISSUE THREAD

**Issue by Tlepsh64** [Tlepsh64/Udacity-Machine-Learning-Foundations-Exercises#1]
#!/usr/bin/python

import matplotlib.pyplot as plt
from prep_terrain_data import makeTerrainData
from class_vis import prettyPicture

features_train, labels_train, features_test, labels_test = makeTerrainData()

### in the scatterplot and identify them visually
grade_fast = [features_train[ii][0] for ii in range(0, len(features_train)) if labels_train[ii]==0]
bumpy_fast = [features_train[ii][1] for ii in range(0, len(features_train)) if labels_train[ii]==0]
grade_slow = [features_train[ii][0] for ii in range(0, len(features_train)) if labels_train[ii]==1]
bumpy_slow = [features_train[ii][1] for ii in range(0, len(features_train)) if labels_train[ii]==1]

#### initial visualization
plt.xlim(0.0, 1.0)
plt.ylim(0.0, 1.0)
plt.scatter(bumpy_fast, grade_fast, color = "b", label="fast")
plt.scatter(grade_slow, bumpy_slow, color = "r", label="slow")
plt.legend()
plt.xlabel("bumpiness")
plt.ylabel("grade")
plt.show()
################################################################################

### visualization code (prettyPicture) to show you the decision boundary
from sklearn import ensemble
from sklearn.metrics import accuracy_score

min_leaf = [50, 100, 150, 200, 250, 300, 350, 400] # Values differ w.r.t to the min_samples_split, but the best ones are btwn 30-60
depth = [3, 50, 100, 150, 200, 250, 300, 350, 400, 450, 500] # Didn't affect the overall performance
n_trees = [10, 20, 30, 40, 50, 70, 90, 120] # Both 20 and 40 are good values
n_leaf_nodes = [5, 10, 15, 20, 30, 40, 45, 50, 65, 80, 95] # Sames as depth
samples_split = [60, 70, 100, 130, 150, 175, 190, 220] # Both 60 and 100 are good
impurity = [0.005, 0.0075, 0.009, 0.01, 0.02, 0.05, 0.08, 0.1] # Best not to play with.. Didnt do so much.
        
clf = ensemble.RandomForestClassifier(min_samples_leaf = 60, min_samples_split = 100, n_estimators = 20, criterion = "entropy", random_state = 5)

t0 = time()    
clf.fit(features_train, labels_train)
print "Training Time:", round(time() - t0, 3), "s"

t1 = time()
preds = clf.predict(features_test)
print "Prediction Time:", round(time() - t1, 3), "s"
    
print "Accuracy:", accuracy_score(labels_test, preds)

"""
Another method, this time I used the simplest of the boosting methods, AdaBoost.

rate = [0.1, 0.2, 0.3, 0.33, 0.35, 0.37, 0.4, 0.5, 0.7, 0.9]

for i in rate:
    clf = ensemble.AdaBoostClassifier(learning_rate = i, n_estimators=100, random_state = 5)
    
    t0 = time()    
    clf.fit(features_train, labels_train)
    print "Training Time:", round(time() - t0, 3), "s"
    
    t1 = time()
    preds = clf.predict(features_test)
    print "Prediction Time:", round(time() - t1, 3), "s"
        
    print "Accuracy for learning rate =", i, ":", accuracy_score(labels_test, preds)"""

try:
    prettyPicture(clf, features_test, labels_test)
except NameError:
    pass
